app/helpers/sessions_helper.rb

```diff
@@ -41,15 +40,9 @@ def log_out
     @current_user = nil
   end
 
-  def add_product_to_cart(product_id)
-    session[:cart][product_id].nil? ? session[:cart][product_id] = 1 : session[:cart][product_id] += 1
-  end
-
   def cart_count
-    session[:cart].values.inject(0) { |sum, value| sum += value }
-  end
+    return 0 unless current_user&.cart.present?
 
-  def get_cart 
-    session[:cart]
+    current_user.cart.count
```

Comment:
I think you could do `current_user&.cart&.count || 0`

Reply:
Very nice, thanks a lot!